packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -55,15 +55,18 @@ final class AssetTransformer {
     required Logger logger,
   }) async {
 
-    String getTempFilePath(int transformStep) {
+    final Directory tempDirectory = _fileSystem.systemTempDirectory.createTempSync();
+
+    File newTempFile(int transformStep) {
```

Comment:
maybe if we change this it should be in another change, but the dependence on the caller keeping track of unique steps seems error prone. Would it be reasonable for this function to keep track of its invocations itself, to ensure even across refactors or new or removed invocations this int monotonically increases?